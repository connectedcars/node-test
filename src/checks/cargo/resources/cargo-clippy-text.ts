import { CargoMessage } from '../cargo-types'

export const cargoClippyFailedOutput: CargoMessage[] = [
  {
    reason: 'compiler-message',
    package_id: 'can 0.1.0 (path+file:///app/can)',
    target: {
      kind: ['lib'],
      crate_types: ['lib'],
      name: 'can',
      src_path: '/app/can/src/lib.rs',
      edition: '2018',
      doctest: true
    },
    message: {
      rendered:
        'error: this comparison involving the minimum or maximum element for this type contains a case that is always true or always false\n   --> can/src/can_session.rs:391:12\n    |\n391 |         if 100 > i32::MAX {}\n    |            ^^^^^^^^^^^^^^\n    |\n    = note: `-D clippy::absurd-extreme-comparisons` implied by `-D clippy::all`\n    = help: because `i32::MAX` is the maximum value for this type, this comparison is always false\n    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#absurd_extreme_comparisons\n\n',
      children: [
        {
          children: [],
          code: null,
          level: 'note',
          message: '`-D clippy::absurd-extreme-comparisons` implied by `-D clippy::all`',
          rendered: null,
          spans: []
        },
        {
          children: [],
          code: null,
          level: 'help',
          message: 'because `i32::MAX` is the maximum value for this type, this comparison is always false',
          rendered: null,
          spans: []
        },
        {
          children: [],
          code: null,
          level: 'help',
          message:
            'for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#absurd_extreme_comparisons',
          rendered: null,
          spans: []
        }
      ],
      code: {
        code: 'clippy::absurd_extreme_comparisons',
        explanation: null
      },
      level: 'error',
      message:
        'this comparison involving the minimum or maximum element for this type contains a case that is always true or always false',
      spans: [
        {
          byte_end: 13338,
          byte_start: 13324,
          column_end: 26,
          column_start: 12,
          expansion: null,
          file_name: 'can/src/can_session.rs',
          is_primary: true,
          label: null,
          line_end: 391,
          line_start: 391,
          suggested_replacement: null,
          suggestion_applicability: null,
          text: [
            {
              highlight_end: 26,
              highlight_start: 12,
              text: '        if 100 > i32::MAX {}'
            }
          ]
        }
      ]
    }
  },
  {
    reason: 'compiler-message',
    package_id: 'can 0.1.0 (path+file:///app/can)',
    target: {
      kind: ['lib'],
      crate_types: ['lib'],
      name: 'can',
      src_path: '/app/can/src/lib.rs',
      edition: '2018',
      doctest: true
    },
    message: {
      rendered: 'error: aborting due to previous error\n\n',
      children: [],
      code: null,
      level: 'error',
      message: 'aborting due to previous error',
      spans: []
    }
  }
]
