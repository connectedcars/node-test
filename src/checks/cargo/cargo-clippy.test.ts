import { cargoClippyCheck } from './cargo-clippy'
import { cargoClippyFailedOutput } from './resources/cargo-clippy-text'

describe('checks/clippy', () => {
  it('converts successful clippy', () => {
    const result = cargoClippyCheck({ data: [], sha: '11963e3cb7ecbb9247f638cc0fb047173a62cf7a' })
    expect(result).toEqual({
      name: 'cargo clippy',
      head_sha: '11963e3cb7ecbb9247f638cc0fb047173a62cf7a',
      completed_at: expect.any(String),
      conclusion: 'success',
      status: 'completed',
      output: {
        title: 'Found no errors',
        summary: 'Found no errors',
        annotations: []
      }
    })
  })
  it('converts failed clippy', () => {
    const result = cargoClippyCheck({ data: cargoClippyFailedOutput, sha: '11963e3cb7ecbb9247f638cc0fb047173a62cf7a' })

    expect(result).toEqual({
      name: 'cargo clippy',
      head_sha: '11963e3cb7ecbb9247f638cc0fb047173a62cf7a',
      completed_at: expect.any(String),
      conclusion: 'failure',
      status: 'completed',
      output: {
        title: 'Total of 2 errors',
        summary: 'Total of 2 errors',
        text: expect.any(String),
        annotations: [
          {
            path: 'can/src/lib.rs',
            start_line: 391,
            end_line: 391,
            annotation_level: 'failure',
            message:
              'this comparison involving the minimum or maximum element for this type contains a case that is always true or always false',
            raw_details: ''
          },
          {
            path: 'can/src/lib.rs',
            start_line: 0,
            end_line: 0,
            annotation_level: 'failure',
            message: 'aborting due to previous error',
            raw_details: ''
          }
        ]
      }
    })
  })
  it('converts buggy clippy', () => {
    const result = cargoClippyCheck({
      data: {} as any,
      sha: '11963e3cb7ecbb9247f638cc0fb047173a62cf7a'
    })
    expect(result).toEqual({
      conclusion: 'neutral',
      completed_at: expect.any(String),
      head_sha: '11963e3cb7ecbb9247f638cc0fb047173a62cf7a',
      name: 'cargo clippy',
      status: 'completed',
      output: {
        title: 'Unexpected clippy output',
        summary: ''
      }
    })
  })
})
