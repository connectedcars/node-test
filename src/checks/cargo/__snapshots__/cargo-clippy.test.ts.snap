// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks/cargo-clippy converts buggy cargo-clippy 1`] = `
Object {
  "completed_at": Any<String>,
  "conclusion": "skipped",
  "head_sha": "11963e3cb7ecbb9247f638cc0fb047173a62cf7a",
  "name": "cargo-clippy",
  "output": Object {
    "summary": "",
    "title": "Unexpected clippy output",
  },
  "status": "completed",
}
`;

exports[`checks/cargo-clippy converts failed cargo-clippy 1`] = `
Object {
  "completed_at": Any<String>,
  "conclusion": "failure",
  "head_sha": "11963e3cb7ecbb9247f638cc0fb047173a62cf7a",
  "name": "cargo-clippy",
  "output": Object {
    "annotations": Array [
      Object {
        "annotation_level": "failure",
        "end_line": 33,
        "message": "error: this looks like you are trying to swap \`foo\` and \`bar\`
  --> src/main.rs:32:5
   |
32 | /     foo = bar;
33 | |     bar = foo;
   | |_____________^ help: try: \`std::mem::swap(&mut foo, &mut bar)\`
   |
   = note: \`#[deny(clippy::almost_swapped)]\` on by default
   = note: or maybe you should use \`std::mem::replace\`?
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#almost_swapped

",
        "path": "src/main.rs",
        "raw_details": "{
    \\"rendered\\": \\"error: this looks like you are trying to swap \`foo\` and \`bar\`\\\\n  --> src/main.rs:32:5\\\\n   |\\\\n32 | /     foo = bar;\\\\n33 | |     bar = foo;\\\\n   | |_____________^ help: try: \`std::mem::swap(&mut foo, &mut bar)\`\\\\n   |\\\\n   = note: \`#[deny(clippy::almost_swapped)]\` on by default\\\\n   = note: or maybe you should use \`std::mem::replace\`?\\\\n   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#almost_swapped\\\\n\\\\n\\",
    \\"children\\": [
        {
            \\"code\\": null,
            \\"level\\": \\"note\\",
            \\"message\\": \\"\`#[deny(clippy::almost_swapped)]\` on by default\\",
            \\"rendered\\": null,
            \\"spans\\": []
        },
        {
            \\"code\\": null,
            \\"level\\": \\"note\\",
            \\"message\\": \\"or maybe you should use \`std::mem::replace\`?\\",
            \\"rendered\\": null,
            \\"spans\\": []
        },
        {
            \\"code\\": null,
            \\"level\\": \\"help\\",
            \\"message\\": \\"for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#almost_swapped\\",
            \\"rendered\\": null,
            \\"spans\\": []
        },
        {
            \\"code\\": null,
            \\"level\\": \\"help\\",
            \\"message\\": \\"try\\",
            \\"rendered\\": null,
            \\"spans\\": [
                {
                    \\"byte_end\\": 411,
                    \\"byte_start\\": 387,
                    \\"column_end\\": 14,
                    \\"column_start\\": 5,
                    \\"expansion\\": null,
                    \\"file_name\\": \\"src/main.rs\\",
                    \\"is_primary\\": true,
                    \\"label\\": null,
                    \\"line_end\\": 33,
                    \\"line_start\\": 32,
                    \\"suggested_replacement\\": \\"std::mem::swap(&mut foo, &mut bar)\\",
                    \\"suggestion_applicability\\": \\"MaybeIncorrect\\",
                    \\"text\\": [
                        {
                            \\"highlight_end\\": 15,
                            \\"highlight_start\\": 5,
                            \\"text\\": \\"    foo = bar;\\"
                        },
                        {
                            \\"highlight_end\\": 14,
                            \\"highlight_start\\": 1,
                            \\"text\\": \\"    bar = foo;\\"
                        }
                    ]
                }
            ]
        }
    ],
    \\"code\\": {
        \\"code\\": \\"clippy::almost_swapped\\",
        \\"explanation\\": null
    },
    \\"level\\": \\"error\\",
    \\"message\\": \\"this looks like you are trying to swap \`foo\` and \`bar\`\\",
    \\"spans\\": [
        {
            \\"byte_end\\": 411,
            \\"byte_start\\": 387,
            \\"column_end\\": 14,
            \\"column_start\\": 5,
            \\"expansion\\": null,
            \\"file_name\\": \\"src/main.rs\\",
            \\"is_primary\\": true,
            \\"label\\": null,
            \\"line_end\\": 33,
            \\"line_start\\": 32,
            \\"suggested_replacement\\": null,
            \\"suggestion_applicability\\": null,
            \\"text\\": [
                {
                    \\"highlight_end\\": 15,
                    \\"highlight_start\\": 5,
                    \\"text\\": \\"    foo = bar;\\"
                },
                {
                    \\"highlight_end\\": 14,
                    \\"highlight_start\\": 1,
                    \\"text\\": \\"    bar = foo;\\"
                }
            ]
        }
    ]
}",
        "start_line": 32,
        "title": "this looks like you are trying to swap \`foo\` and \`bar\`",
      },
      Object {
        "annotation_level": "failure",
        "end_line": 15,
        "message": "warning: use of a blacklisted/placeholder name \`foo\`
  --> src/main.rs:15:11
   |
15 |       let foo = foo();
   |           ^^^
   |
   = note: \`#[warn(clippy::blacklisted_name)]\` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#blacklisted_name

",
        "path": "src/main.rs",
        "raw_details": "{
    \\"rendered\\": \\"warning: use of a blacklisted/placeholder name \`foo\`\\\\n  --> src/main.rs:15:11\\\\n   |\\\\n15 |       let foo = foo();\\\\n   |           ^^^\\\\n   |\\\\n   = note: \`#[warn(clippy::blacklisted_name)]\` on by default\\\\n   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#blacklisted_name\\\\n\\\\n\\",
    \\"children\\": [
        {
            \\"code\\": null,
            \\"level\\": \\"note\\",
            \\"message\\": \\"\`#[warn(clippy::blacklisted_name)]\` on by default\\",
            \\"rendered\\": null,
            \\"spans\\": []
        },
        {
            \\"code\\": null,
            \\"level\\": \\"help\\",
            \\"message\\": \\"for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#blacklisted_name\\",
            \\"rendered\\": null,
            \\"spans\\": []
        }
    ],
    \\"code\\": {
        \\"code\\": \\"clippy::blacklisted_name\\",
        \\"explanation\\": null
    },
    \\"level\\": \\"warning\\",
    \\"message\\": \\"use of a blacklisted/placeholder name \`foo\`\\",
    \\"spans\\": [
        {
            \\"byte_end\\": 160,
            \\"byte_start\\": 157,
            \\"column_end\\": 14,
            \\"column_start\\": 11,
            \\"expansion\\": null,
            \\"file_name\\": \\"src/main.rs\\",
            \\"is_primary\\": true,
            \\"label\\": null,
            \\"line_end\\": 15,
            \\"line_start\\": 15,
            \\"suggested_replacement\\": null,
            \\"suggestion_applicability\\": null,
            \\"text\\": [
                {
                    \\"highlight_end\\": 14,
                    \\"highlight_start\\": 11,
                    \\"text\\": \\"      let foo = foo();\\"
                }
            ]
        }
    ]
}",
        "start_line": 15,
        "title": "use of a blacklisted/placeholder name \`foo\`",
      },
      Object {
        "annotation_level": "failure",
        "end_line": 29,
        "message": "warning: use of a blacklisted/placeholder name \`foo\`
  --> src/main.rs:29:13
   |
29 |     let mut foo = 0;
   |             ^^^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#blacklisted_name

",
        "path": "src/main.rs",
        "raw_details": "{
    \\"rendered\\": \\"warning: use of a blacklisted/placeholder name \`foo\`\\\\n  --> src/main.rs:29:13\\\\n   |\\\\n29 |     let mut foo = 0;\\\\n   |             ^^^\\\\n   |\\\\n   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#blacklisted_name\\\\n\\\\n\\",
    \\"children\\": [
        {
            \\"code\\": null,
            \\"level\\": \\"help\\",
            \\"message\\": \\"for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#blacklisted_name\\",
            \\"rendered\\": null,
            \\"spans\\": []
        }
    ],
    \\"code\\": {
        \\"code\\": \\"clippy::blacklisted_name\\",
        \\"explanation\\": null
    },
    \\"level\\": \\"warning\\",
    \\"message\\": \\"use of a blacklisted/placeholder name \`foo\`\\",
    \\"spans\\": [
        {
            \\"byte_end\\": 355,
            \\"byte_start\\": 352,
            \\"column_end\\": 16,
            \\"column_start\\": 13,
            \\"expansion\\": null,
            \\"file_name\\": \\"src/main.rs\\",
            \\"is_primary\\": true,
            \\"label\\": null,
            \\"line_end\\": 29,
            \\"line_start\\": 29,
            \\"suggested_replacement\\": null,
            \\"suggestion_applicability\\": null,
            \\"text\\": [
                {
                    \\"highlight_end\\": 16,
                    \\"highlight_start\\": 13,
                    \\"text\\": \\"    let mut foo = 0;\\"
                }
            ]
        }
    ]
}",
        "start_line": 29,
        "title": "use of a blacklisted/placeholder name \`foo\`",
      },
      Object {
        "annotation_level": "failure",
        "end_line": 37,
        "message": "warning: in an \`if\` condition, avoid complex blocks or closures with blocks; instead, move the block or closure higher and bind it with a \`let\`
  --> src/main.rs:37:5
   |
37 |     if { let b = false; b } {}
   |     ^^^^^^^^^^^^^^^^^^^^^^^ help: try: \`let res = { let b = false; b }; if res\`
   |
   = note: \`#[warn(clippy::blocks_in_if_conditions)]\` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#blocks_in_if_conditions

",
        "path": "src/main.rs",
        "raw_details": "{
    \\"rendered\\": \\"warning: in an \`if\` condition, avoid complex blocks or closures with blocks; instead, move the block or closure higher and bind it with a \`let\`\\\\n  --> src/main.rs:37:5\\\\n   |\\\\n37 |     if { let b = false; b } {}\\\\n   |     ^^^^^^^^^^^^^^^^^^^^^^^ help: try: \`let res = { let b = false; b }; if res\`\\\\n   |\\\\n   = note: \`#[warn(clippy::blocks_in_if_conditions)]\` on by default\\\\n   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#blocks_in_if_conditions\\\\n\\\\n\\",
    \\"children\\": [
        {
            \\"code\\": null,
            \\"level\\": \\"note\\",
            \\"message\\": \\"\`#[warn(clippy::blocks_in_if_conditions)]\` on by default\\",
            \\"rendered\\": null,
            \\"spans\\": []
        },
        {
            \\"code\\": null,
            \\"level\\": \\"help\\",
            \\"message\\": \\"for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#blocks_in_if_conditions\\",
            \\"rendered\\": null,
            \\"spans\\": []
        },
        {
            \\"code\\": null,
            \\"level\\": \\"help\\",
            \\"message\\": \\"try\\",
            \\"rendered\\": null,
            \\"spans\\": [
                {
                    \\"byte_end\\": 443,
                    \\"byte_start\\": 420,
                    \\"column_end\\": 28,
                    \\"column_start\\": 5,
                    \\"expansion\\": null,
                    \\"file_name\\": \\"src/main.rs\\",
                    \\"is_primary\\": true,
                    \\"label\\": null,
                    \\"line_end\\": 37,
                    \\"line_start\\": 37,
                    \\"suggested_replacement\\": \\"let res = { let b = false; b }; if res\\",
                    \\"suggestion_applicability\\": \\"MachineApplicable\\",
                    \\"text\\": [
                        {
                            \\"highlight_end\\": 28,
                            \\"highlight_start\\": 5,
                            \\"text\\": \\"    if { let b = false; b } {}\\"
                        }
                    ]
                }
            ]
        }
    ],
    \\"code\\": {
        \\"code\\": \\"clippy::blocks_in_if_conditions\\",
        \\"explanation\\": null
    },
    \\"level\\": \\"warning\\",
    \\"message\\": \\"in an \`if\` condition, avoid complex blocks or closures with blocks; instead, move the block or closure higher and bind it with a \`let\`\\",
    \\"spans\\": [
        {
            \\"byte_end\\": 443,
            \\"byte_start\\": 420,
            \\"column_end\\": 28,
            \\"column_start\\": 5,
            \\"expansion\\": null,
            \\"file_name\\": \\"src/main.rs\\",
            \\"is_primary\\": true,
            \\"label\\": null,
            \\"line_end\\": 37,
            \\"line_start\\": 37,
            \\"suggested_replacement\\": null,
            \\"suggestion_applicability\\": null,
            \\"text\\": [
                {
                    \\"highlight_end\\": 28,
                    \\"highlight_start\\": 5,
                    \\"text\\": \\"    if { let b = false; b } {}\\"
                }
            ]
        }
    ]
}",
        "start_line": 37,
        "title": "in an \`if\` condition, avoid complex blocks or closures with blocks; instead, move the block or closure higher and bind it with a \`let\`",
      },
      Object {
        "annotation_level": "failure",
        "end_line": 37,
        "message": "warning: returning the result of a \`let\` binding from a block
  --> src/main.rs:37:25
   |
37 |     if { let b = false; b } {}
   |          -------------- ^
   |          |
   |          unnecessary \`let\` binding
   |
   = note: \`#[warn(clippy::let_and_return)]\` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#let_and_return
help: return the expression directly
   |
37 |     if {  false } {}
   |         --^^^^^

",
        "path": "src/main.rs",
        "raw_details": "{
    \\"rendered\\": \\"warning: returning the result of a \`let\` binding from a block\\\\n  --> src/main.rs:37:25\\\\n   |\\\\n37 |     if { let b = false; b } {}\\\\n   |          -------------- ^\\\\n   |          |\\\\n   |          unnecessary \`let\` binding\\\\n   |\\\\n   = note: \`#[warn(clippy::let_and_return)]\` on by default\\\\n   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#let_and_return\\\\nhelp: return the expression directly\\\\n   |\\\\n37 |     if {  false } {}\\\\n   |         --^^^^^\\\\n\\\\n\\",
    \\"children\\": [
        {
            \\"code\\": null,
            \\"level\\": \\"note\\",
            \\"message\\": \\"\`#[warn(clippy::let_and_return)]\` on by default\\",
            \\"rendered\\": null,
            \\"spans\\": []
        },
        {
            \\"code\\": null,
            \\"level\\": \\"help\\",
            \\"message\\": \\"for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#let_and_return\\",
            \\"rendered\\": null,
            \\"spans\\": []
        },
        {
            \\"code\\": null,
            \\"level\\": \\"help\\",
            \\"message\\": \\"return the expression directly\\",
            \\"rendered\\": null,
            \\"spans\\": [
                {
                    \\"byte_end\\": 439,
                    \\"byte_start\\": 425,
                    \\"column_end\\": 24,
                    \\"column_start\\": 10,
                    \\"expansion\\": null,
                    \\"file_name\\": \\"src/main.rs\\",
                    \\"is_primary\\": true,
                    \\"label\\": null,
                    \\"line_end\\": 37,
                    \\"line_start\\": 37,
                    \\"suggested_replacement\\": \\"\\",
                    \\"suggestion_applicability\\": \\"MachineApplicable\\",
                    \\"text\\": [
                        {
                            \\"highlight_end\\": 24,
                            \\"highlight_start\\": 10,
                            \\"text\\": \\"    if { let b = false; b } {}\\"
                        }
                    ]
                },
                {
                    \\"byte_end\\": 441,
                    \\"byte_start\\": 440,
                    \\"column_end\\": 26,
                    \\"column_start\\": 25,
                    \\"expansion\\": null,
                    \\"file_name\\": \\"src/main.rs\\",
                    \\"is_primary\\": true,
                    \\"label\\": null,
                    \\"line_end\\": 37,
                    \\"line_start\\": 37,
                    \\"suggested_replacement\\": \\"false\\",
                    \\"suggestion_applicability\\": \\"MachineApplicable\\",
                    \\"text\\": [
                        {
                            \\"highlight_end\\": 26,
                            \\"highlight_start\\": 25,
                            \\"text\\": \\"    if { let b = false; b } {}\\"
                        }
                    ]
                }
            ]
        }
    ],
    \\"code\\": {
        \\"code\\": \\"clippy::let_and_return\\",
        \\"explanation\\": null
    },
    \\"level\\": \\"warning\\",
    \\"message\\": \\"returning the result of a \`let\` binding from a block\\",
    \\"spans\\": [
        {
            \\"byte_end\\": 439,
            \\"byte_start\\": 425,
            \\"column_end\\": 24,
            \\"column_start\\": 10,
            \\"expansion\\": null,
            \\"file_name\\": \\"src/main.rs\\",
            \\"is_primary\\": false,
            \\"label\\": \\"unnecessary \`let\` binding\\",
            \\"line_end\\": 37,
            \\"line_start\\": 37,
            \\"suggested_replacement\\": null,
            \\"suggestion_applicability\\": null,
            \\"text\\": [
                {
                    \\"highlight_end\\": 24,
                    \\"highlight_start\\": 10,
                    \\"text\\": \\"    if { let b = false; b } {}\\"
                }
            ]
        },
        {
            \\"byte_end\\": 441,
            \\"byte_start\\": 440,
            \\"column_end\\": 26,
            \\"column_start\\": 25,
            \\"expansion\\": null,
            \\"file_name\\": \\"src/main.rs\\",
            \\"is_primary\\": true,
            \\"label\\": null,
            \\"line_end\\": 37,
            \\"line_start\\": 37,
            \\"suggested_replacement\\": null,
            \\"suggestion_applicability\\": null,
            \\"text\\": [
                {
                    \\"highlight_end\\": 26,
                    \\"highlight_start\\": 25,
                    \\"text\\": \\"    if { let b = false; b } {}\\"
                }
            ]
        }
    ]
}",
        "start_line": 37,
        "title": "returning the result of a \`let\` binding from a block",
      },
      Object {
        "annotation_level": "failure",
        "end_line": 51,
        "message": "warning: \`assert!(true)\` will be optimized out by the compiler
  --> src/main.rs:51:9
   |
51 |         assert!(true);
   |         ^^^^^^^^^^^^^^
   |
   = note: \`#[warn(clippy::assertions_on_constants)]\` on by default
   = help: remove it
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#assertions_on_constants
   = note: this warning originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

",
        "path": "src/main.rs",
        "raw_details": "{
    \\"rendered\\": \\"warning: \`assert!(true)\` will be optimized out by the compiler\\\\n  --> src/main.rs:51:9\\\\n   |\\\\n51 |         assert!(true);\\\\n   |         ^^^^^^^^^^^^^^\\\\n   |\\\\n   = note: \`#[warn(clippy::assertions_on_constants)]\` on by default\\\\n   = help: remove it\\\\n   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#assertions_on_constants\\\\n   = note: this warning originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)\\\\n\\\\n\\",
    \\"children\\": [
        {
            \\"code\\": null,
            \\"level\\": \\"note\\",
            \\"message\\": \\"\`#[warn(clippy::assertions_on_constants)]\` on by default\\",
            \\"rendered\\": null,
            \\"spans\\": []
        },
        {
            \\"code\\": null,
            \\"level\\": \\"help\\",
            \\"message\\": \\"remove it\\",
            \\"rendered\\": null,
            \\"spans\\": []
        },
        {
            \\"code\\": null,
            \\"level\\": \\"help\\",
            \\"message\\": \\"for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#assertions_on_constants\\",
            \\"rendered\\": null,
            \\"spans\\": []
        }
    ],
    \\"code\\": {
        \\"code\\": \\"clippy::assertions_on_constants\\",
        \\"explanation\\": null
    },
    \\"level\\": \\"warning\\",
    \\"message\\": \\"\`assert!(true)\` will be optimized out by the compiler\\",
    \\"spans\\": [
        {
            \\"byte_end\\": 603,
            \\"byte_start\\": 589,
            \\"column_end\\": 23,
            \\"column_start\\": 9,
            \\"expansion\\": {
                \\"def_site_span\\": {
                    \\"byte_end\\": 41339,
                    \\"byte_start\\": 41183,
                    \\"column_end\\": 6,
                    \\"column_start\\": 5,
                    \\"expansion\\": null,
                    \\"file_name\\": \\"/home/user/.rustup/toolchains/1.49.0-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/macros/mod.rs\\",
                    \\"is_primary\\": false,
                    \\"label\\": null,
                    \\"line_end\\": 1221,
                    \\"line_start\\": 1218,
                    \\"suggested_replacement\\": null,
                    \\"suggestion_applicability\\": null,
                    \\"text\\": [
                        {
                            \\"highlight_end\\": 1,
                            \\"highlight_start\\": 5,
                            \\"text\\": \\"    macro_rules! assert {\\"
                        },
                        {
                            \\"highlight_end\\": 1,
                            \\"highlight_start\\": 1,
                            \\"text\\": \\"        ($cond:expr $(,)?) => {{ /* compiler built-in */ }};\\"
                        },
                        {
                            \\"highlight_end\\": 1,
                            \\"highlight_start\\": 1,
                            \\"text\\": \\"        ($cond:expr, $($arg:tt)+) => {{ /* compiler built-in */ }};\\"
                        },
                        {
                            \\"highlight_end\\": 6,
                            \\"highlight_start\\": 1,
                            \\"text\\": \\"    }\\"
                        }
                    ]
                },
                \\"macro_decl_name\\": \\"assert!\\",
                \\"span\\": {
                    \\"byte_end\\": 603,
                    \\"byte_start\\": 589,
                    \\"column_end\\": 23,
                    \\"column_start\\": 9,
                    \\"expansion\\": null,
                    \\"file_name\\": \\"src/main.rs\\",
                    \\"is_primary\\": false,
                    \\"label\\": null,
                    \\"line_end\\": 51,
                    \\"line_start\\": 51,
                    \\"suggested_replacement\\": null,
                    \\"suggestion_applicability\\": null,
                    \\"text\\": [
                        {
                            \\"highlight_end\\": 23,
                            \\"highlight_start\\": 9,
                            \\"text\\": \\"        assert!(true);\\"
                        }
                    ]
                }
            },
            \\"file_name\\": \\"src/main.rs\\",
            \\"is_primary\\": true,
            \\"label\\": null,
            \\"line_end\\": 51,
            \\"line_start\\": 51,
            \\"suggested_replacement\\": null,
            \\"suggestion_applicability\\": null,
            \\"text\\": [
                {
                    \\"highlight_end\\": 23,
                    \\"highlight_start\\": 9,
                    \\"text\\": \\"        assert!(true);\\"
                }
            ]
        }
    ]
}",
        "start_line": 51,
        "title": "\`assert!(true)\` will be optimized out by the compiler",
      },
      Object {
        "annotation_level": "failure",
        "end_line": 56,
        "message": "warning: \`assert!(false)\` should probably be replaced
  --> src/main.rs:56:5
   |
56 |     assert!(false);
   |     ^^^^^^^^^^^^^^^
   |
   = help: use \`panic!()\` or \`unreachable!()\`
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#assertions_on_constants
   = note: this warning originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

",
        "path": "src/main.rs",
        "raw_details": "{
    \\"rendered\\": \\"warning: \`assert!(false)\` should probably be replaced\\\\n  --> src/main.rs:56:5\\\\n   |\\\\n56 |     assert!(false);\\\\n   |     ^^^^^^^^^^^^^^^\\\\n   |\\\\n   = help: use \`panic!()\` or \`unreachable!()\`\\\\n   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#assertions_on_constants\\\\n   = note: this warning originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)\\\\n\\\\n\\",
    \\"children\\": [
        {
            \\"code\\": null,
            \\"level\\": \\"help\\",
            \\"message\\": \\"use \`panic!()\` or \`unreachable!()\`\\",
            \\"rendered\\": null,
            \\"spans\\": []
        },
        {
            \\"code\\": null,
            \\"level\\": \\"help\\",
            \\"message\\": \\"for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#assertions_on_constants\\",
            \\"rendered\\": null,
            \\"spans\\": []
        }
    ],
    \\"code\\": {
        \\"code\\": \\"clippy::assertions_on_constants\\",
        \\"explanation\\": null
    },
    \\"level\\": \\"warning\\",
    \\"message\\": \\"\`assert!(false)\` should probably be replaced\\",
    \\"spans\\": [
        {
            \\"byte_end\\": 660,
            \\"byte_start\\": 645,
            \\"column_end\\": 20,
            \\"column_start\\": 5,
            \\"expansion\\": {
                \\"def_site_span\\": {
                    \\"byte_end\\": 41339,
                    \\"byte_start\\": 41183,
                    \\"column_end\\": 6,
                    \\"column_start\\": 5,
                    \\"expansion\\": null,
                    \\"file_name\\": \\"/home/user/.rustup/toolchains/1.49.0-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/macros/mod.rs\\",
                    \\"is_primary\\": false,
                    \\"label\\": null,
                    \\"line_end\\": 1221,
                    \\"line_start\\": 1218,
                    \\"suggested_replacement\\": null,
                    \\"suggestion_applicability\\": null,
                    \\"text\\": [
                        {
                            \\"highlight_end\\": 26,
                            \\"highlight_start\\": 5,
                            \\"text\\": \\"    macro_rules! assert {\\"
                        },
                        {
                            \\"highlight_end\\": 61,
                            \\"highlight_start\\": 1,
                            \\"text\\": \\"        ($cond:expr $(,)?) => {{ /* compiler built-in */ }};\\"
                        },
                        {
                            \\"highlight_end\\": 68,
                            \\"highlight_start\\": 1,
                            \\"text\\": \\"        ($cond:expr, $($arg:tt)+) => {{ /* compiler built-in */ }};\\"
                        },
                        {
                            \\"highlight_end\\": 6,
                            \\"highlight_start\\": 1,
                            \\"text\\": \\"    }\\"
                        }
                    ]
                },
                \\"macro_decl_name\\": \\"assert!\\",
                \\"span\\": {
                    \\"byte_end\\": 660,
                    \\"byte_start\\": 645,
                    \\"column_end\\": 20,
                    \\"column_start\\": 5,
                    \\"expansion\\": null,
                    \\"file_name\\": \\"src/main.rs\\",
                    \\"is_primary\\": false,
                    \\"label\\": null,
                    \\"line_end\\": 56,
                    \\"line_start\\": 56,
                    \\"suggested_replacement\\": null,
                    \\"suggestion_applicability\\": null,
                    \\"text\\": [
                        {
                            \\"highlight_end\\": 20,
                            \\"highlight_start\\": 5,
                            \\"text\\": \\"    assert!(false);\\"
                        }
                    ]
                }
            },
            \\"file_name\\": \\"src/main.rs\\",
            \\"is_primary\\": true,
            \\"label\\": null,
            \\"line_end\\": 56,
            \\"line_start\\": 56,
            \\"suggested_replacement\\": null,
            \\"suggestion_applicability\\": null,
            \\"text\\": [
                {
                    \\"highlight_end\\": 20,
                    \\"highlight_start\\": 5,
                    \\"text\\": \\"    assert!(false);\\"
                }
            ]
        }
    ]
}",
        "start_line": 56,
        "title": "\`assert!(false)\` should probably be replaced",
      },
    ],
    "summary": "Total of 7 issues",
    "text": "## Results

| Message level           | Amount |
| ----------------------- | ------ |
| Internal compiler error |      0 |
| Error                   |      2 |
| Warning                 |     10 |
| Note                    |      0 |
| Help                    |      0 |",
    "title": "Total of 7 issues",
  },
  "status": "completed",
}
`;

exports[`checks/cargo-clippy converts successful cargo-clippy 1`] = `
Object {
  "completed_at": Any<String>,
  "conclusion": "success",
  "head_sha": "11963e3cb7ecbb9247f638cc0fb047173a62cf7a",
  "name": "cargo-clippy",
  "output": Object {
    "annotations": Array [],
    "summary": "Found no issues",
    "title": "Found no issues",
  },
  "status": "completed",
}
`;
